test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.test.shared
+
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.RealmQuery
+import io.realm.RealmResults
+import io.realm.entities.Sample
+import io.realm.internal.platform.singleThreadDispatcher
+import io.realm.test.platform.PlatformUtils
+import kotlinx.coroutines.async
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.runBlocking
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+
+class QueryTests {
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .path("$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun find() {
+        realm.query(Sample::class)
+            .find()
+            .let { results -> assertEquals(0, results.size) }
+
+        val stringValue = "some string"
+        realm.writeBlocking {
+            // Queries inside a write transaction are live and can be reused within the closure
+            val query = query(Sample::class)
+
+            assertEquals(0, query.find().size)
+            copyToRealm(Sample().apply { stringField = stringValue })
+            assertEquals(1, query.find().size)
+        }
+
+        // No filter
+        realm.query(Sample::class)
```

Comment:
Any reason to assert the same as the first one in this test?

Reply:
It isn't quite the same assert. Here we should receive a result, thus the `assertEquals(1, results.size)` whereas in the first one we have no results since it's before the write transaction that populates the Realm.